packages/core/r4b/data-types/boolean.ts

```diff
@@ -0,0 +1,88 @@
+import { merge } from "lodash";
+/**
+ * A boolean
+ *
+ * @see https://hl7.org/fhir/datatypes.html#boolean
+ */
+
+export interface FhirBooleanFormatOptions {
+  labels?: {
+    true?: string;
+    false?: string;
+    null?: string;
+    undefined?: string;
+  };
+}
+
+export interface FhirBooleanTypeAdapter {
+  locale: string | undefined;
+  /**
+   * Parse a FHIR boolean
+   *
+   * @see https://hl7.org/fhir/datatypes.html#boolean
+   */
+  parse(value: string | boolean | null | undefined): boolean | undefined;
+
+  /**
+   * Format a FHIR boolean
+   *
+   * @see https://hl7.org/fhir/datatypes.html#boolean
+   */
+  format(
+    value: string | boolean | null | undefined,
+    options?: FhirBooleanFormatOptions
+  ): string;
+}
+
+const fhirBooleanRegexp = new RegExp("^(true)|(false)$");
+const defaultLabels = {
+  true: "true",
+  false: "false",
+  null: "",
+  undefined: "",
+};
+
+/**
+ * Return a {@link FhirBooleanTypeAdapter}
+ */
+export function fhirBooleanTypeAdapter(
+  locale: string | undefined
+): FhirBooleanTypeAdapter {
+  return {
+    locale,
+    parse(value) {
+      if (typeof value === "boolean") return value;
+
+      const sanitizedValue = value?.trim()?.toLowerCase();
+      if (!sanitizedValue) {
+        return undefined;
+      }
+
+      // value is a string
+      if (!sanitizedValue.match(fhirBooleanRegexp))
+        throw new Error(
+          "Value does not match the fhir boolean format as described in `https://hl7.org/fhir/datatypes.html#boolean'"
+        );
+
+      return sanitizedValue === "true";
+    },
+
+    format(value, options) {
+      const fhirBoolean =
+        typeof value === "boolean" ? value : this.parse(value);
+
+      options ||= {};
+      const labels = merge({}, defaultLabels, options.labels || {});
+      switch (fhirBoolean) {
+        case true:
+          return labels["true"];
+        case false:
+          return labels["false"];
+        case null:
+          return labels["null"];
+        case undefined:
+          return labels["undefined"];
+      }
```

Comment:
Javascript is happy to coerce strings when using templates :-) 

```suggestion
      return labels[`${fhirBoolean}`];
```

Reply:
ho Not sure if TS will be happy

I was using `String(fhirBoolean)` and he was not happy